server/src/controller/habitController.js

```diff
@@ -87,11 +94,13 @@ const deleteHabit = async (req, res) => {
     const habit = await Habit.findOneAndDelete({ _id: id });
 
     if (!habit) {
+      //send 404 Not Found status code for getHabit fail (habit is null)
       return res.status(404).json({ error: "habit not found" });
     }
 
     res.status(200).json(habit);
   } catch (err) {
+    //send 500 Internal Server Error status code for getAllHabits fail
```

Comment:
You are sending a 400 below

Reply:
Oh! forgot to edit here. Sorry will update it in next commit